api/doc.go

```diff
@@ -15,72 +15,57 @@
 // structure and the various payloads that can be sent and received.
 //
-// The proxy protocol is composed of messages: requests and responses. These
-// form a small RPC protocol, requests being similar to a function call and
-// responses encoding the result of the call.
+// The proxy protocol is composed of commands, responses and notifications.
+// They all share the same frame structure: a header followed by an optional
+// payload.
 //
-// Each message is composed of a header and some optional data:
+// • Commands are always initiated by a client, never by the proxy itself.
 //
-//  ┌────────────────┬────────────────┬──────────────────────────────┐
-//  │  Data Length   │    Reserved    │  Data (request or response)  │
-//  │   (32 bits)    │    (32 bits)   │     (data length bytes)      │
-//  └────────────────┴────────────────┴──────────────────────────────┘
+// • Responses are sent by the proxy to acknowledge commands.
 //
-// - Data Length is in bytes and encoded in network order.
+// • Notifications are sent by either the proxy or clients and do not generate
+// responses.
 //
-// - Reserved is reserved for future use.
+// The frame format is illustrated below:
 //
-// - Data is the JSON-encoded request or response data
+//                      1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 3 3
+//  0 1 2 3 4 5 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+//  ┌───────────────────────────┬───────────────┬───────────────┐
+//  │          Version          │ Header Length │   Reserved    │
+//  ├───────────────────────────┼───────┬───────┼───────────────┤
+//  │          Reserved         │ Res.  │ Type  │    Opcode     │
+//  ├───────────────────────────┴───────┴───────┴───────────────┤
+//  │                      Payload Length                       │
+//  ├───────────────────────────────────────────────────────────┤
+//  │                                                           │
+//  │                         Payload                           │
+//  │                                                           │
+//  │      (variable length, optional and opcode-specific)      │
+//  │                                                           │
+//  └───────────────────────────────────────────────────────────┘
 //
-// On top of of this request/response mechanism, the proxy defines payloads,
-// which are effectively the various function calls defined in the API.
+// All header fields are encoded in network order (big endian).
 //
-// Requests have 2 fields: the payload id (function name) and its data
-// (function argument(s))
+// • Version (16 bits) is the proxy protocol version. See api.Version for
+// details about what information it encodes.
 //
-//  type Request struct {
-//      Id    string          `json:"id"`
-//      Data *json.RawMessage `json:"data,omitempty"`
-//  }
+// • Header Length (8 bits) is the length of the header in number of 32 bits
+// words.  Header Length is greater or equal to 3 (12 bytes).
```

Comment:
Right, frame+header_length point at payload_length bytes, will add the clarification.